Mathlib/RingTheory/HahnSeries/Addition.lean

```diff
@@ -78,17 +78,55 @@ theorem support_add_subset {x y : HahnSeries Γ R} : support (x + y) ⊆ support
   rw [ha.1, ha.2, add_zero]
 #align hahn_series.support_add_subset HahnSeries.support_add_subset
 
+protected theorem min_le_min_add {Γ} [LinearOrder Γ] {x y : HahnSeries Γ R} (hx : x ≠ 0)
+    (hy : y ≠ 0) (hxy : x + y ≠ 0) :
+    min (Set.IsWF.min x.isWF_support (support_nonempty_iff.2 hx))
+      (Set.IsWF.min y.isWF_support (support_nonempty_iff.2 hy)) ≤
+      Set.IsWF.min (x + y).isWF_support (support_nonempty_iff.2 hxy) := by
+  rw [← Set.IsWF.min_union]
+  exact Set.IsWF.min_le_min_of_subset (support_add_subset (x := x) (y := y))
+
+theorem min_orderTop_le_orderTop_add {Γ} [LinearOrder Γ] {x y : HahnSeries Γ R} :
+    min x.orderTop y.orderTop ≤ (x + y).orderTop := by
+  by_cases hx : x = 0; · simp [hx]
+  by_cases hy : y = 0; · simp [hy]
+  by_cases hxy : x + y = 0; · simp [hxy]
+  rw [orderTop_of_ne hx, orderTop_of_ne hy, orderTop_of_ne hxy, ← @WithTop.coe_min,
+    WithTop.coe_le_coe]
+  exact HahnSeries.min_le_min_add hx hy hxy
+
 theorem min_order_le_order_add {Γ} [Zero Γ] [LinearOrder Γ] {x y : HahnSeries Γ R}
     (hxy : x + y ≠ 0) : min x.order y.order ≤ (x + y).order := by
   by_cases hx : x = 0; · simp [hx]
   by_cases hy : y = 0; · simp [hy]
   rw [order_of_ne hx, order_of_ne hy, order_of_ne hxy]
-  apply le_of_eq_of_le _ (Set.IsWF.min_le_min_of_subset (support_add_subset (x := x) (y := y)))
-  · simp
-  · simp [hy]
-  · exact (Set.IsWF.min_union _ _ _ _).symm
+  exact HahnSeries.min_le_min_add hx hy hxy
 #align hahn_series.min_order_le_order_add HahnSeries.min_order_le_order_add
 
+theorem orderTop_add_eq {Γ} [LinearOrder Γ] {x y : HahnSeries Γ R}
+    (hxy : x.orderTop < y.orderTop) : (x + y).orderTop = x.orderTop := by
```

Comment:
Would it be worth having `_left` and `_right` variants, one with `x` and one with `y` on the right hand side? Also for the next theorem.

Reply:
I have added the variants.  Unfortunately, I was not able to avoid basically duplicating the proofs, since addition is not commutative here.